### PR TITLE
Fix AllocationRoutedStepTests.testExecuteAllocateNotComplete

### LINKED ISSUE #73095: [CI] AllocationRoutedStepTests testExecuteAllocateNotComplete failure
**Build scan**: https://gradle-enterprise.elastic.co/s/3pt3xaajwp4ys

**Repro line**: `./gradlew ':x-pack:plugin:core:test' --tests "org.elasticsearch.xpack.core.ilm.AllocationRoutedStepTests.testExecuteAllocateNotComplete" -Dtests.seed=66DD601BB5C6EB6F -Dtests.locale=es-BO -Dtests.timezone=Asia/Beirut -Druntime.java=11`

**Reproduces locally?**: Yes

**Applicable branches**: `master`

**Failure excerpt**:

```
org.elasticsearch.xpack.core.ilm.AllocationRoutedStepTests > testExecuteAllocateNotComplete FAILED
    java.lang.AssertionError: Node roles must not be provided as attributes but saw attributes {XmRX=lGjfrfsPimOAciX, EmRxOHENSz=IaR, FCmGvSmVumyWwMwRlkS=bO, FmHOcUh=mpeejIGnHVip, GSkYgnb=ShKHhqTABIyKVQg, zhQapvDLvRaQAHGF=Ztr, XlQcWgyhl=JIKiStIQekkkJq, hyo=gwGtEOAf, ml=dFYLAoXQaDEeSdRE}
        at __randomizedtesting.SeedInfo.seed([66DD601BB5C6EB6F:9EF6008EA87B3447]:0)
        at org.elasticsearch.cluster.node.DiscoveryNode.<init>(DiscoveryNode.java:204)
        at org.elasticsearch.cluster.node.DiscoveryNode.<init>(DiscoveryNode.java:164)
        at org.elasticsearch.cluster.node.DiscoveryNode.createLocal(DiscoveryNode.java:213)
        at org.elasticsearch.xpack.core.ilm.AllocationRoutedStepTests.assertAllocateStatus(AllocationRoutedStepTests.java:396)
        at org.elasticsearch.xpack.core.ilm.AllocationRoutedStepTests.testExecuteAllocateNotComplete(AllocationRoutedStepTests.java:263)
```

### PR DESCRIPTION
**Pull request by andreidan**
The randomMap of allocation routings that we generated could include a
key that was not a random attribute name but a node role name (eg. `ml`)

Closes #73095

### CODE REVIEW

**Comment by andreidan** [comment]
@elasticmachine update branch

**Comment by andreidan** [comment]
@elasticmachine run elasticsearch-ci/part-2

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/AllocateActionTests.java [hunk 1]
```diff
@@ -103,19 +104,21 @@ public void testAllMapsNullOrEmpty() {
...
         assertEquals("[" + AllocateAction.NUMBER_OF_REPLICAS_FIELD.getPreferredName() + "] must be >= 0", exception.getMessage());
     }
 
-    public static Map<String, String> randomMap(int minEntries, int maxEntries) {
+    public static Map<String, String> randomAllocationRoutingMap(int minEntries, int maxEntries) {
         Map<String, String> map = new HashMap<>();
         int numIncludes = randomIntBetween(minEntries, maxEntries);
         for (int i = 0; i < numIncludes; i++) {
-            map.put(randomAlphaOfLengthBetween(2, 20), randomAlphaOfLengthBetween(2, 20));
+            String attributeName = randomValueOtherThanMany(generatedName -> DiscoveryNodeRole.roleNames().contains(generatedName),
```
Comment (probakowski) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/AllocateActionTests.java:119]:
```suggestion
            String attributeName = randomValueOtherThanMany(DiscoveryNodeRole.roleNames()::contains,
```

**Review by probakowski — APPROVED** [review]
LGTM 👍  thanks Andrei!

**Comment by andreidan** [comment]
opened https://github.com/elastic/elasticsearch/issues/74953

**Comment by andreidan** [comment]
@elasticmachine run elasticsearch-ci/rest-compatibility

**Comment by andreidan** [comment]
@elasticmachine update branch

**Pull request merged**